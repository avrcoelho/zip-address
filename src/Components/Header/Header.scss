@import '../../Scss/root';

$RainbowSize: 700px;

@keyframes magaluRainbowMove {
	from { background-position: 0; }
	to { background-position: $RainbowSize 0; }
}

.header {
	height: $header-height;
	position: relative;

	&:after {
		@include luiza-gradient;
		background-size: $RainbowSize;
		bottom: -5px;
		content: '';
		display: block;
		height: 5px;
		position: absolute;
		width: 100%;
	}

	&--loading:after { animation: magaluRainbowMove 1s linear infinite }

	&__logo {
		@include square(100px);
		@include hover-shadow;
		border-radius: 50%;
		display: inline-block;
		font-size: 0;
		line-height: 0;
		overflow: hidden;
		position: relative;
		transform: translateY(25%);
		z-index: 1;
	}
}