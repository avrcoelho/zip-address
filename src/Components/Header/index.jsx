import React from 'react';

import { useAppContext } from '../../Providers/App.Context';

import logo from './luizalabs-logo.png';
import './Header.scss';

export default function Header() {
	const { loading } = useAppContext();

	return (
		<header className={`${loading ? 'header--loading ' : ''}header bg-secondary color-primary text-center`}>
			<h1 className="my-0">
				<a
					className="header__logo"
					href="http://www.cabecadelab.com.br/about.html"
					target="_blank"
				>
					LuizaLabs
					<img
						alt="Luiza Labs"
						className="header__logo-image full-width"
						src={logo}
						title="Luiza Labs"
					/>
				</a>
			</h1>
		</header>
	)
};