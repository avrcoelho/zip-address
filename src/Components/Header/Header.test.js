import React from 'react';
import { mount } from 'enzyme';

import AppProvider from '../../Providers/App.Context'

import Header from './index';

describe('<Header />', () => {
	it('should mount properly', () => {
		const wrapper = mount(
			<AppProvider>
				<Header />
			</AppProvider>
		);

		expect(wrapper.find('header')).toMatchSnapshot();
	});

	it('shoud add loader class when context loading true', () => {
		const wrapper = mount(
			<AppProvider initialState={true}>
				<Header />
			</AppProvider>
		);

		expect(wrapper.find('header').hasClass('header--loading')).toBeTruthy()
	});
});