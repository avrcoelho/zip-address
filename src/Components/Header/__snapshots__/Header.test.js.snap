// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should mount properly 1`] = `
<header
  className="header bg-secondary color-primary text-center"
>
  <h1
    className="my-0"
  >
    <a
      className="header__logo"
      href="http://www.luizalabs.com/"
      target="_blank"
    >
      LuizaLabs
      <img
        alt="Luiza Labs"
        className="header__logo-image full-width"
        src="test-file-stub"
        title="Luiza Labs"
      />
    </a>
  </h1>
</header>
`;
